File: levelordertree.py
# Time Complexity : O(N)
# Space Complexity : O(N)
# Did this code successfully run on Leetcode : Yes
# Any problem you faced while coding this : No


# Your code here along with comments explaining your approach

from collections import deque
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        
        result = []
        
        if root == None:
            return result
        
        queue = deque()
        queue.append(root)
        
        while queue:
            count = len(queue)
            levelelements = []
            
            for i in range(count):
                element = queue.popleft()
                levelelements.append(element.val)
                
                if element.left!=None:
                    queue.append(element.left)
                
                if element.right!=None:
                    queue.append(element.right)
            
            result.append(levelelements)
            
        return result
           
                    
                
                
            
                
            
                
                
                
                
        
        
        
        
        

File: LevelOrderTree.java
/* Time Complexity : O(n)
   Space Complexity : O(n) where n is number of nodes   
   Did this code successfully run on Leetcode : Yes
   Any problem you faced while coding this : No
*/
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if(root==null) return new ArrayList<>();
        List<List<Integer>> res = new ArrayList<>();
        Queue<TreeNode> q=new LinkedList<>();
        q.add(root);
        while(!q.isEmpty()){
            int size=q.size();
            List<Integer> li = new ArrayList<>();
            for(int i=0;i<size;i++)
            {
                TreeNode node = q.poll();
                li.add(node.val);
                if(node.left!=null)
                    q.add(node.left);
                if(node.right!=null)
                    q.add(node.right);
            }
            res.add(li);
        }
        return res;
    }
}